tests/modules/images/image-loading-optimization/class-ilo-html-tag-processor-tests.php

```diff
@@ -0,0 +1,333 @@
+					'/*[0][self::HTML]/*[1][self::BODY]/*[0][self::DL]',
+					'/*[0][self::HTML]/*[1][self::BODY]/*[1][self::P]',
+					'/*[0][self::HTML]/*[1][self::BODY]/*[0][self::FIELDSET]',
+					'/*[0][self::HTML]/*[1][self::BODY]/*[1][self::P]',
+					'/*[0][self::HTML]/*[1][self::BODY]/*[0][self::FIGCAPTION]',
+					'/*[0][self::HTML]/*[1][self::BODY]/*[1][self::P]',
+					'/*[0][self::HTML]/*[1][self::BODY]/*[0][self::FIGURE]',
+					'/*[0][self::HTML]/*[1][self::BODY]/*[1][self::P]',
+					'/*[0][self::HTML]/*[1][self::BODY]/*[0][self::FOOTER]',
+					'/*[0][self::HTML]/*[1][self::BODY]/*[1][self::P]',
+					'/*[0][self::HTML]/*[1][self::BODY]/*[0][self::FORM]',
+					'/*[0][self::HTML]/*[1][self::BODY]/*[1][self::P]',
+					'/*[0][self::HTML]/*[1][self::BODY]/*[0][self::H1]',
+					'/*[0][self::HTML]/*[1][self::BODY]/*[1][self::P]',
+					'/*[0][self::HTML]/*[1][self::BODY]/*[0][self::H2]',
+					'/*[0][self::HTML]/*[1][self::BODY]/*[1][self::P]',
+					'/*[0][self::HTML]/*[1][self::BODY]/*[0][self::H3]',
+					'/*[0][self::HTML]/*[1][self::BODY]/*[1][self::P]',
+					'/*[0][self::HTML]/*[1][self::BODY]/*[0][self::H4]',
+					'/*[0][self::HTML]/*[1][self::BODY]/*[1][self::P]',
+					'/*[0][self::HTML]/*[1][self::BODY]/*[0][self::H5]',
+					'/*[0][self::HTML]/*[1][self::BODY]/*[1][self::P]',
+					'/*[0][self::HTML]/*[1][self::BODY]/*[0][self::H6]',
+					'/*[0][self::HTML]/*[1][self::BODY]/*[1][self::P]',
+					'/*[0][self::HTML]/*[1][self::BODY]/*[0][self::HEADER]',
+					'/*[0][self::HTML]/*[1][self::BODY]/*[1][self::P]',
+					'/*[0][self::HTML]/*[1][self::BODY]/*[0][self::HGROUP]',
+					'/*[0][self::HTML]/*[1][self::BODY]/*[1][self::P]',
+					'/*[0][self::HTML]/*[1][self::BODY]/*[0][self::HR]',
+					'/*[0][self::HTML]/*[1][self::BODY]/*[1][self::P]',
+					'/*[0][self::HTML]/*[1][self::BODY]/*[0][self::MAIN]',
+					'/*[0][self::HTML]/*[1][self::BODY]/*[1][self::P]',
+					'/*[0][self::HTML]/*[1][self::BODY]/*[0][self::MENU]',
+					'/*[0][self::HTML]/*[1][self::BODY]/*[1][self::P]',
+					'/*[0][self::HTML]/*[1][self::BODY]/*[0][self::NAV]',
+					'/*[0][self::HTML]/*[1][self::BODY]/*[1][self::P]',
+					'/*[0][self::HTML]/*[1][self::BODY]/*[0][self::OL]',
+					'/*[0][self::HTML]/*[1][self::BODY]/*[1][self::P]',
+					'/*[0][self::HTML]/*[1][self::BODY]/*[0][self::PRE]',
+					'/*[0][self::HTML]/*[1][self::BODY]/*[1][self::P]',
+					'/*[0][self::HTML]/*[1][self::BODY]/*[0][self::SEARCH]',
+					'/*[0][self::HTML]/*[1][self::BODY]/*[1][self::P]',
+					'/*[0][self::HTML]/*[1][self::BODY]/*[0][self::SECTION]',
+					'/*[0][self::HTML]/*[1][self::BODY]/*[1][self::P]',
+					'/*[0][self::HTML]/*[1][self::BODY]/*[0][self::TABLE]',
+					'/*[0][self::HTML]/*[1][self::BODY]/*[1][self::P]',
+					'/*[0][self::HTML]/*[1][self::BODY]/*[0][self::UL]',
+				),
+			),
+		);
+	}
+
+	/**
+	 * Test open_tags() and get_xpath().
+	 *
+	 * @covers ::open_tags
+	 * @covers ::get_xpath
+	 *
+	 * @dataProvider data_provider_sample_documents
```

Comment:
Nice if we add `@param` that used in `dataProvider`.

Reply:
Yeah, somewhat. But it looks like most of the other tests using data providers don't have `@param` tags in the plugin. Not that this is a good reason.